test/local/redis.js

```diff
@@ -0,0 +1,141 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const LIB_DIR = '../../lib'
+
+const { assert } = require('chai')
+const error = require(`${LIB_DIR}/error`)
+const mocks = require('../mocks')
+const P = require(`${LIB_DIR}/promise`)
+const proxyquire = require('proxyquire')
+const sinon = require('sinon')
+
+describe('lib/redis:', () => {
```

Comment:
It was the wrong call to only have integration tests for this module, because they make harder to exercise the error-handling logic. So I took the opportunity to add unit tests for it here.